### ISSUE TITLE
Add iris 2 support

### ISSUE THREAD

**Issue by edisonf** [uvdl/yocto-ornl#3]

**Comment by edisonf** [comment]
@bvacaliuc 

Branch [feature/iris2-support](https://github.com/uvdl/yocto-ornl/tree/feature/iris2-support) currently contains a new device tree called imx6q-iris2.dtb

The kernel configuration  can be found at: meta-ornl/recipes-kernel/linux/linux-variscite/var-som-mx6-ornl/defconfig 

At this point, if you want to boot using this new device tree you can do:

1. Stop at uboot
2. Set fdt_file variable to be imx6q-iris2.dtb
<pre>
setenv fdt_file imx6q-iris2.dtb
</pre>

3. Boot the board

<pre>
boot
</pre>

I added everything from [this commit](https://github.com/uvdl/linux-imx/commit/cf3abdee50e01b6da80f7fc1c8dc8a3d4de44786)  except for the TC358743 node.

**Comment by bvacaliuc** [comment]
I went thru the procedure described [in the branch README.md](https://github.com/uvdl/yocto-ornl/blob/feature/iris2-support/README.md) and it appears as if it only rebuilt the kernel uImage and dtb file, not the rootfs.  Also, the link to the .sdcard described below seems to be connected to just the root filesystem, not a bootable disk:

```
ubuntu@ip-172-31-21-219:~/ornl-dart-yocto/build_ornl$ ls -al tmp/deploy/images/var-som-mx6-ornl
total 3110592
drwxr-xr-x 2 ubuntu ubuntu       4096 Dec 20 16:09 .
drwxrwxr-x 3 ubuntu ubuntu       4096 Dec 15 15:23 ..
-rw-r--r-- 1 ubuntu ubuntu 1593835520 Dec 15 16:01 fsl-image-ornl-var-som-mx6-ornl-20181215150959.rootfs.ext4
-rw-r--r-- 1 ubuntu ubuntu      55271 Dec 15 16:01 fsl-image-ornl-var-som-mx6-ornl-20181215150959.rootfs.manifest
-rw-r--r-- 1 ubuntu ubuntu 1610612736 Dec 15 16:01 fsl-image-ornl-var-som-mx6-ornl-20181215150959.rootfs.sdcard
-rw-r--r-- 1 ubuntu ubuntu  438837683 Dec 15 16:01 fsl-image-ornl-var-som-mx6-ornl-20181215150959.rootfs.tar.gz
-rw-r--r-- 1 ubuntu ubuntu     143620 Dec 15 16:01 fsl-image-ornl-var-som-mx6-ornl-20181215150959.testdata.json
-rw-r--r-- 1 ubuntu ubuntu     143637 Dec 20 16:09 fsl-image-ornl-var-som-mx6-ornl-20181220160532.testdata.json
lrwxrwxrwx 1 ubuntu ubuntu         58 Dec 15 16:01 fsl-image-ornl-var-som-mx6-ornl.ext4 -> fsl-image-ornl-var-som-mx6-ornl-20181215150959.rootfs.ext4
lrwxrwxrwx 1 ubuntu ubuntu         62 Dec 15 16:01 fsl-image-ornl-var-som-mx6-ornl.manifest -> fsl-image-ornl-var-som-mx6-ornl-20181215150959.rootfs.manifest
lrwxrwxrwx 1 ubuntu ubuntu         60 Dec 15 16:01 fsl-image-ornl-var-som-mx6-ornl.sdcard -> fsl-image-ornl-var-som-mx6-ornl-20181215150959.rootfs.sdcard
lrwxrwxrwx 1 ubuntu ubuntu         60 Dec 15 16:01 fsl-image-ornl-var-som-mx6-ornl.tar.gz -> fsl-image-ornl-var-som-mx6-ornl-20181215150959.rootfs.tar.gz
lrwxrwxrwx 1 ubuntu ubuntu         60 Dec 20 16:09 fsl-image-ornl-var-som-mx6-ornl.testdata.json -> fsl-image-ornl-var-som-mx6-ornl-20181220160532.testdata.json
-rw-rw-r-- 2 ubuntu ubuntu     808489 Dec 20 16:07 modules--4.1.15-r0-var-som-mx6-ornl-20181220160532.tgz
lrwxrwxrwx 2 ubuntu ubuntu         54 Dec 20 16:07 modules-var-som-mx6-ornl.tgz -> modules--4.1.15-r0-var-som-mx6-ornl-20181220160532.tgz
lrwxrwxrwx 2 ubuntu ubuntu         45 Dec 15 15:24 SPL -> SPL-var-som-mx6-ornl-2015.04-r0-sd-2015.04-r0
lrwxrwxrwx 2 ubuntu ubuntu         47 Dec 15 15:24 SPL-nand -> SPL-var-som-mx6-ornl-2015.04-r0-nand-2015.04-r0
lrwxrwxrwx 2 ubuntu ubuntu         45 Dec 15 15:24 SPL-sd -> SPL-var-som-mx6-ornl-2015.04-r0-sd-2015.04-r0
lrwxrwxrwx 2 ubuntu ubuntu         45 Dec 15 15:24 SPL-var-som-mx6-ornl -> SPL-var-som-mx6-ornl-2015.04-r0-sd-2015.04-r0
-rw-r--r-- 2 ubuntu ubuntu      31744 Dec 15 15:24 SPL-var-som-mx6-ornl-2015.04-r0-nand-2015.04-r0
-rw-r--r-- 2 ubuntu ubuntu      31744 Dec 15 15:24 SPL-var-som-mx6-ornl-2015.04-r0-sd-2015.04-r0
lrwxrwxrwx 2 ubuntu ubuntu         47 Dec 15 15:24 SPL-var-som-mx6-ornl-nand -> SPL-var-som-mx6-ornl-2015.04-r0-nand-2015.04-r0
lrwxrwxrwx 2 ubuntu ubuntu         45 Dec 15 15:24 SPL-var-som-mx6-ornl-sd -> SPL-var-som-mx6-ornl-2015.04-r0-sd-2015.04-r0
lrwxrwxrwx 2 ubuntu ubuntu         24 Dec 15 15:24 u-boot.img -> u-boot-sd-2015.04-r0.img
lrwxrwxrwx 2 ubuntu ubuntu         26 Dec 15 15:24 u-boot.img-nand -> u-boot-nand-2015.04-r0.img
lrwxrwxrwx 2 ubuntu ubuntu         24 Dec 15 15:24 u-boot.img-sd -> u-boot-sd-2015.04-r0.img
-rw-r--r-- 2 ubuntu ubuntu     450844 Dec 15 15:24 u-boot-nand-2015.04-r0.img
-rw-r--r-- 2 ubuntu ubuntu     316008 Dec 15 15:24 u-boot-sd-2015.04-r0.img
lrwxrwxrwx 2 ubuntu ubuntu         24 Dec 15 15:24 u-boot-var-som-mx6-ornl.img -> u-boot-sd-2015.04-r0.img
lrwxrwxrwx 2 ubuntu ubuntu         26 Dec 15 15:24 u-boot-var-som-mx6-ornl.img-nand -> u-boot-nand-2015.04-r0.img
lrwxrwxrwx 2 ubuntu ubuntu         24 Dec 15 15:24 u-boot-var-som-mx6-ornl.img-sd -> u-boot-sd-2015.04-r0.img
lrwxrwxrwx 2 ubuntu ubuntu         53 Dec 20 16:07 uImage -> uImage--4.1.15-r0-var-som-mx6-ornl-20181220160532.bin
-rw-r--r-- 2 ubuntu ubuntu      41746 Dec 20 16:07 uImage--4.1.15-r0-imx6dl-var-som-vsc-20181220160532.dtb
-rw-r--r-- 2 ubuntu ubuntu      42381 Dec 20 16:07 uImage--4.1.15-r0-imx6q-iris2-20181220160532.dtb
-rw-r--r-- 2 ubuntu ubuntu      42365 Dec 20 16:07 uImage--4.1.15-r0-imx6q-var-dart-20181220160532.dtb
-rw-r--r-- 2 ubuntu ubuntu      42133 Dec 20 16:07 uImage--4.1.15-r0-imx6q-var-som-cap-20181220160532.dtb
-rw-r--r-- 2 ubuntu ubuntu      42546 Dec 20 16:07 uImage--4.1.15-r0-imx6q-var-som-res-20181220160532.dtb
-rw-r--r-- 2 ubuntu ubuntu      43179 Dec 20 16:07 uImage--4.1.15-r0-imx6q-var-som-vsc-20181220160532.dtb
-rw-r--r-- 2 ubuntu ubuntu    6208208 Dec 20 16:07 uImage--4.1.15-r0-var-som-mx6-ornl-20181220160532.bin
lrwxrwxrwx 2 ubuntu ubuntu         55 Dec 20 16:07 uImage-imx6dl-var-som-vsc.dtb -> uImage--4.1.15-r0-imx6dl-var-som-vsc-20181220160532.dtb
lrwxrwxrwx 2 ubuntu ubuntu         48 Dec 20 16:07 uImage-imx6q-iris2.dtb -> uImage--4.1.15-r0-imx6q-iris2-20181220160532.dtb
lrwxrwxrwx 2 ubuntu ubuntu         51 Dec 20 16:07 uImage-imx6q-var-dart.dtb -> uImage--4.1.15-r0-imx6q-var-dart-20181220160532.dtb
lrwxrwxrwx 2 ubuntu ubuntu         54 Dec 20 16:07 uImage-imx6q-var-som-cap.dtb -> uImage--4.1.15-r0-imx6q-var-som-cap-20181220160532.dtb
lrwxrwxrwx 2 ubuntu ubuntu         54 Dec 20 16:07 uImage-imx6q-var-som-res.dtb -> uImage--4.1.15-r0-imx6q-var-som-res-20181220160532.dtb
lrwxrwxrwx 2 ubuntu ubuntu         54 Dec 20 16:07 uImage-imx6q-var-som-vsc.dtb -> uImage--4.1.15-r0-imx6q-var-som-vsc-20181220160532.dtb
lrwxrwxrwx 2 ubuntu ubuntu         53 Dec 20 16:07 uImage-var-som-mx6-ornl.bin -> uImage--4.1.15-r0-var-som-mx6-ornl-20181220160532.bin
```

I would like to see a Makefile or something to generate a bootable SD card that can be used to evaluate the new kernel on to the board.  The way the debian branch did it was pretty ideal:  It would [create an sdcard image](https://github.com/uvdl/debian-var/blob/iris2/make_var_som_mx6_debian.sh#L896) that included uboot, a VFAT partition containing the .dtb file and uImage and a EXT4 partition containing the root filesystem that expanded to the size of the target SD card.  _(I edited it to limit it to 4GB so it wouldn't take so long to upload, but that was just me.)_

**Comment by bvacaliuc** [comment]
An example might be in this [Makefile](https://github.com/uvdl/yocto-ornl/blob/project/bogdan/Makefile). 

I guess all the parts are there, its just confusing which parts have to go where.  I.e. How much to reserve for u-boot? is 8192 sectors (4MB) enough?  Will it boot a VFAT partition containing `uImage-imx6q-iris2.dtb` and `uImage`, or will it require the full decorated names?  Can we create an EXT4 partition and expand `fsl-image-ornl-var-som-mx6-ornl-20181215150959.rootfs.tar.gz` onto it?  See also [this issue](https://github.com/uvdl/meta/issues/1).

**Comment by bvacaliuc** [comment]
per Edison's recommendation, I deleted `~/ornl-dart-yocto/build_ornl/tmp` and ran bitbake.  This time it did regenerate the files.

Upon testing, on the DT6CustomBoard, with the switch set to SD, it did boot ok.  However, upon inserting into the Iris2 R0 board, with the switch set to SD, it **did not boot**.  What I had to do was to copy block 1 thru 8191 from the debian SD card image into the yocto card block 1 thru 8191.  **Then it booted**.  Something is strange with the u-boot loader generated by yocto.
